Platform/UTMJailbreak.m

```diff
@@ -56,15 +82,99 @@ static bool am_i_being_debugged() {
+    
+    // Inspired by codesign.c in Darwin sources for Security.framework
+    
+    // Find our mach-o header
+    Dl_info dl_info;
+    if (dladdr(app_entitlements, &dl_info) == 0)
+        return nil;
+    if (dl_info.dli_fbase == NULL)
+        return nil;
+    char *base = dl_info.dli_fbase;
+    struct mach_header_64 *header = dl_info.dli_fbase;
+    if (header->magic != MH_MAGIC_64)
+        return nil;
+    
+    // Simulator executables have fake entitlements in the code signature. The real entitlements can be found in an __entitlements section.
+    size_t entitlements_size;
+    uint8_t *entitlements_data = getsectiondata(header, "__TEXT", "__entitlements", &entitlements_size);
+    if (entitlements_data != NULL) {
+        NSData *data = [NSData dataWithBytesNoCopy:entitlements_data
+                                            length:entitlements_size
+                                      freeWhenDone:NO];
+        return entitlements = [NSPropertyListSerialization propertyListWithData:data
+                                                                        options:NSPropertyListImmutable
+                                                                         format:nil
+                                                                          error:nil];
+    }
+    
+    // Find the LC_CODE_SIGNATURE
+    struct load_command *lc = (void *) (base + sizeof(*header));
+    struct linkedit_data_command *cs_lc = NULL;
+    for (uint32_t i = 0; i < header->ncmds; i++) {
+        if (lc->cmd == LC_CODE_SIGNATURE) {
+            cs_lc = (void *) lc;
+            break;
+        }
+        lc = (void *) ((char *) lc + lc->cmdsize);
+    }
+    if (cs_lc == NULL)
+        return nil;
+
+    // Read the code signature off disk, as it's apparently not loaded into memory
+    NSFileHandle *fileHandle = [NSFileHandle fileHandleForReadingFromURL:NSBundle.mainBundle.executableURL error:nil];
+    if (fileHandle == nil)
+        return nil;
+    [fileHandle seekToFileOffset:cs_lc->dataoff];
+    NSData *csData = [fileHandle readDataOfLength:cs_lc->datasize];
+    [fileHandle closeFile];
+    const struct cs_superblob *cs = csData.bytes;
+    if (ntohl(cs->magic) != 0xfade0cc0)
+        return nil;
+    
+    // Find the entitlements in the code signature
+    for (uint32_t i = 0; i < ntohl(cs->count); i++) {
+        struct cs_entitlements *ents = (void *) ((char *) cs + ntohl(cs->index[i].offset));
+        if (ntohl(ents->magic) == 0xfade7171) {
+            return parse_entitlements(ents->entitlements, ntohl(ents->length) - offsetof(struct cs_entitlements, entitlements));
+        }
+    }
+    return nil;
+}
 
 bool jb_has_jit_entitlement(void) {
-    void *addr = mmap(NULL, PAGE_SIZE, PROT_READ | PROT_WRITE | PROT_EXEC, MAP_ANON | MAP_PRIVATE | MAP_JIT, -1, 0);
-    if (addr != MAP_FAILED) {
-        munmap(addr, PAGE_SIZE);
-        return true;
-    } else {
-        return false;
+    static NSDictionary *entitlements = nil;
```

Comment:
static variables are zero-initialized by default, FWIW.

Reply:
I know but I want to be explicit here. Sometimes readability is more important than succinctness.

However, this does mean the code I copy pasted at line 108 should be changed as well.